common/rdm/AckTimerResponder.cpp

```diff
@@ -208,13 +208,29 @@ RDMResponse *AckTimerResponder::ResponseFromQueuedMessage(
     const class QueuedResponse *queued_response) {
   switch (queued_response->CommandClass()) {
     case RDMCommand::GET_COMMAND_RESPONSE:
+      return new RDMGetResponse(
+        // coverity[SWAPPED_ARGUMENTS]
+        request->DestinationUID(),
+        request->SourceUID(),
+        request->TransactionNumber(),
+        RDM_ACK,
+        QueuedMessageCount(),
+        ROOT_RDM_DEVICE,
+        queued_response->Pid(),
+        queued_response->ParamData(),
+        queued_response->ParamDataSize());
     case RDMCommand::SET_COMMAND_RESPONSE:
-      return GetResponseFromData(request,
```

Comment:
Ah, so it was just that it had the wrong PID in the response? It looks like GetResponseWithPid will be a neater fit, or does it have to come from the root device? Of course in that case you could extend GetResponseWithPid to have a version that passes in the subdevice too.